.gitpod.yml

```
@@ -0,0 +1,14 @@
+tasks:
+  - name: Configure nerdctl
+    command: mkdir -p /home/gitpod/.config/nerdctl && echo 'snapshotter = "native"' > /home/gitpod/.config/nerdctl/nerdctl.toml
+  - name: Run containerd
+    command: bash -c "curl https://raw.githubusercontent.com/containerd/containerd/main/script/setup/install-cni | bash" && sudo containerd
```

`nerdctl run` failed for me because `containerd`'s root directory was outside of `/workspace`. Only `/workspace` provides a "direct" filesystem (e.g. xfs), all other paths are on `shiftfs` which breaks a number of `runc`/`containerd` assumptions.

Explicitly placing the `root` directory under `/workspace` does the trick and `nerdctl run --rm -it docker.io/library/alpine:latest` works.

```suggestion
  - name: Run containerd
    init: mkdir -p /workspace/containerd/root
    command: bash -c "curl https://raw.githubusercontent.com/containerd/containerd/main/script/setup/install-cni | bash" && sudo containerd --root /workspace/containerd/root
```

---

> Explicitly placing the `root` directory under `/workspace` does the trick and `nerdctl run --rm -it docker.io/library/alpine:latest` works.

That worked before, this doesn't:

```
$ nerdctl run -it --rm -p 3000:80 nginx
FATA[0000] failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: time="2022-11-07T17:46:34Z" level=fatal msg="failed to call cni.Setup: plugin type=\"portmap\" failed (add): unable to enable route_localnet: open /proc/sys/net/ipv4/conf/nerdctl0/route_localnet: read-only file system"
Failed to write to log, write /var/lib/nerdctl/1935db59/containers/default/2e39d1142c21a4dd17a782b001b214e0c4143b6d24753116fba43dd66a7902fc/oci-hook.createRuntime.log: file already closed: unknown
```